pandas/tests/io/excel/test_writers.py

```diff
@@ -1162,6 +1162,19 @@ def test_path_local_path(self, engine, ext):
                                        path="foo.{ext}".format(ext=ext))
         tm.assert_frame_equal(result, df)
 
+    def test_merged_cell_custom_objects(self, engine, ext):
+        # see GH-27006
+        mi = MultiIndex.from_tuples([(pd.Period('2018'), pd.Period('2018Q1')),
+                                     (pd.Period('2018'), pd.Period('2018Q2'))])
+        expected = DataFrame(np.random.rand(2, len(mi)), columns=mi)
```

Comment:
Can use `np.ones((2, 2))` in place of `np.random.rand`. Being overly cautious but no need for random data here

Reply:
done - i had add a dtype='np.float64' to read_excel though, as it was converting the 1's to int64's

Reply:
You can just use `convert_float=True` in read_excel

Reply:
`convert_float` is `True` by default. It works if I set it to `False`, but then the index column that's read back in is treated as a float, and I have to convert it back to int64 before comparing.

Or I can choose NOT to write the index to excel, but I don't really think that's good practice, no?

Reply:
OK no worries. Let's keep this as is then